sensor/common/registry/registry_store_test.go

```diff
@@ -333,28 +418,43 @@ func TestRegistryStore_GenImgIntName(t *testing.T) {
 	tt := []struct {
 		prefix    string
 		namespace string
+		name      string
 		registry  string
 		expected  string
 	}{
-		{"", "", "", ""},
-		{"PRE", "", "", "PRE"},
-		{"PRE", "", "REG", "PRE/reg:REG"},
-		{"PRE", "NAME", "", "PRE/ns:NAME"},
-		{"PRE", "NAME", "REG", "PRE/ns:NAME/reg:REG"},
+		{"", "", "", "", ""},
+		{"PRE", "", "", "", "PRE"},
+		{"PRE", "", "", "REG", "PRE/reg:REG"},
+		{"PRE", "NAMESP", "", "", "PRE/ns:NAMESP"},
+		{"PRE", "NAMESP", "", "REG", "PRE/ns:NAMESP/reg:REG"},
+		{"PRE", "", "NAME", "", "PRE/name:NAME"},
+		{"PRE", "NAMESP", "NAME", "", "PRE/ns:NAMESP/name:NAME"},
+		{"PRE", "NAMESP", "NAME", "REG", "PRE/ns:NAMESP/name:NAME/reg:REG"},
 	}
 
 	for i, test := range tt {
 		t.Run(fmt.Sprintf("%v", i), func(t *testing.T) {
-			actual := genIntegrationName(test.prefix, test.namespace, test.registry)
+			actual := genIntegrationName(test.prefix, test.namespace, test.name, test.registry)
 			assert.Equal(t, test.expected, actual)
 		})
 	}
 }
 
-func TestDataRaceAtCleanup(_ *testing.T) {
-	testNamespace := "test-ns"
+func TestDataRaceAtCleanup(t *testing.T) {
+	t.Run("SecretsByHost", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SensorPullSecretsByName, false)
+		dataRaceAtCleanupSubTest()
+	})
+
+	t.Run("SecretsByName", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SensorPullSecretsByName, true)
+		dataRaceAtCleanupSubTest()
+	})
+}
+
+func dataRaceAtCleanupSubTest() {
```

Comment:
is this testing for a panic or deadlock? How will this test pass/fail?

Reply:
Tests are ran with the `-race` flag, when a race (and maybe deadlock?) is detected the test will panic and fail.